pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -81,6 +82,9 @@ func (s *NodeController) Run(period time.Duration, syncNodeList bool) {
 	// Register intial set of nodes with their status set.
 	var nodes *api.NodeList
 	var err error
+
+	record.StartRecording(s.kubeClient.Events(""), api.EventSource{Component: "nodecontroller"})
```

Comment:
Optional:

```go
defer record.StartRecording(s.kubeClient.Events(""), api.EventSource{Component: "nodecontroller"}).Stop()
```

Will make this "nice" if Run() is called more than once (e.g. for testing or because it crashed or something).